chain/jsonrpc-primitives/src/types/query.rs

```diff
@@ -0,0 +1,151 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcQueryRequest {
+    #[serde(flatten)]
+    pub block_reference: near_primitives::types::BlockReference,
+    #[serde(flatten)]
+    pub request: near_primitives::views::QueryRequest,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcQueryError {
+    #[error("IO Error: {0}")]
+    IOError(String),
+    #[error("There are no fully synchronized blocks yet")]
+    NotSyncedYet,
+    #[error("The node does not track the shard")]
+    DoesNotTrackShard,
+    #[error("Invalid account ID {0}")]
+    InvalidAccount(near_primitives::types::AccountId),
+    #[error("Account ID {0} has never been observed on the node")]
+    AccountDoesNotExist(near_primitives::types::AccountId),
+    #[error("Contract code for contract ID {0} has never been observed on the node")]
+    ContractCodeDoesNotExist(near_primitives::types::AccountId),
+    #[error("Access key for public key {0} has never been observed on the node")]
+    AccessKeyDoesNotExist(String),
+    #[error("VM error occurred: {0}")]
+    VMError(String),
```

Comment:
I suggest we apply the following reasoning:

* Each individual error variant should be self-contained, e.g. `NotSyncedYet`, `DoesNotTrackShard`, and even `VMError` imply a certain context. I suggest we capture the context in the naming:
  * `NoSyncedBlocks` ("There are no fully synchronized blocks on the node yet")
  * `UnavailableShard { requested_shard_id: ShardId }` ("The node does not track the shard #{requested_shard_id} where the requested account resides")
  * `FailedFunctionCall { error_message: String }` - we may consider returning `VMError` structure here since It is what we may want to do in `tx` and `broadcast_tx_*` endpoints anyway, and having a debug-formatted string of the enum is not helpful anyway, so we may just as well just go and dump `VMError`
* I suggest we keep our terminology around "does-not-exist" consistent and always use either "unavailable" or "unknown": `UnknownAccount`, `UnknownAccessKey`
* Contract code can be missing on an account, so it is not part of the "does not exist" crew, but rather just: `NoContractCode { contract_account_id: AccountId }`